server/routes/associatedPrisoner/associatePrisonerPage.ts

```diff
@@ -0,0 +1,237 @@
+    const draftId = getDraftIdFromString(req.params.adjudicationNumber)
+    const { roleCode } = req.params
+
+    setRedirectUrl(req, draftId, roleCode, this.pageOptions.isPreviouslySubmitted())
+
+    return redirectToDeletePersonPage(res, req.body.prisonerId)
+  }
+
+  search = async (req: Request, res: Response): Promise<void> => {
+    const user = res.locals
+    const draftId = getDraftIdFromString(req.params.adjudicationNumber)
+    const { roleCode } = req.params
+    const { selectedAnswerId } = req.body
+    const { prisonerNumber } = await this.readFromApi(draftId, user as User)
+
+    setRedirectUrl(req, draftId, roleCode, this.pageOptions.isPreviouslySubmitted())
+
+    if (!req.body.prisonerSearchNameInput || req.body.prisonerSearchNameInput === '') {
+      const pageData = await this.getDisplayData(
+        {
+          prisonerNumber,
+          currentAssociatedPrisonerNumber: null,
+          currentAssociatedPrisonerName: null,
+        },
+        selectedAnswerId,
+        user as User
+      )
+
+      return renderData(res, draftId, roleCode, pageData, [errors.MISSING_INTERNAL_ASSOCIATED_PRISONER_ASSIST])
+    }
+    return redirectToSearchForPersonPage(res, req.body.prisonerSearchNameInput)
+  }
+
+  readFromApi = async (draftId: number, user: User): Promise<RoleAssociatedPrisoner> => {
+    return extractAssociatedDetails(await this.placeOnReportService.getDraftAdjudicationDetails(draftId, user))
+  }
+
+  saveToApiUpdate = async (
+    draftId: number,
+    roleAssociatedPrisoner: RoleAssociatedPrisoner,
+    currentUser: User
+  ): Promise<DraftAdjudicationResult> => {
+    return this.placeOnReportService.saveAssociatedPrisoner(
+      draftId,
+      {
+        associatedPrisonersNumber: roleAssociatedPrisoner.currentAssociatedPrisonerNumber,
+        associatedPrisonersName: roleAssociatedPrisoner.currentAssociatedPrisonerName,
+      },
+      currentUser
+    )
+  }
+
+  getDisplayData = async (
+    roleAssociatedPrisoner: RoleAssociatedPrisoner,
+    locationOverride: AssociatedPrisonerLocation,
+    currentUser: User
+  ): Promise<DisplayData> => {
+    const prisoner = await this.placeOnReportService.getPrisonerDetails(
+      roleAssociatedPrisoner.prisonerNumber,
+      currentUser
+    )
+    const location = locationOverride == null ? getPrisonerLocation(roleAssociatedPrisoner) : locationOverride
+    let associatedPrisonersName
+
+    switch (location) {
+      case AssociatedPrisonerLocation.EXTERNAL:
+        associatedPrisonersName = roleAssociatedPrisoner.currentAssociatedPrisonerName
+        break
+      case AssociatedPrisonerLocation.INTERNAL:
+        associatedPrisonersName = roleAssociatedPrisoner.currentAssociatedPrisonerNumber
+          ? await this.getCurrentAssociatedPrisonersName(
+              roleAssociatedPrisoner.currentAssociatedPrisonerNumber,
+              currentUser
+            )
+          : null
+        break
+      default:
+    }
```

Comment:
Could we put this in a separate function instead of using `let` and then reassigning?